Tests/UnitTests/Core/Propagator/CovarianceEngineTests.cpp

```diff
@@ -164,5 +171,210 @@ BOOST_AUTO_TEST_CASE(covariance_engine_test) {
   BOOST_CHECK_NE(*(std::get<0>(boundResult).covariance()),
                  Covariance::Identity());
 }
+
+BOOST_AUTO_TEST_CASE(CovarianceConversionTest) {
+  GeometryContext gctx;
+
+  std::size_t n = 100;
+
+  std::mt19937 rng{42};
+
+  std::uniform_real_distribution<double> locDist(-5., 5.);
+  std::uniform_real_distribution<double> globDist(-50., 50.);
+  std::uniform_real_distribution<double> angleDist(-2 * M_PI, 2 * M_PI);
+  std::uniform_real_distribution<double> unitDist(0, 1);
+
+  for (std::size_t b = 0; b < 10; b++) {
+    const Vector3 bField =
+        Vector3(unitDist(rng), unitDist(rng), unitDist(rng)) * 3_T;
+
+    for (std::size_t j = 0; j < 10; j++) {
+      Transform3 transformA = Transform3::Identity();
+      transformA = AngleAxis3(angleDist(rng), Vector3::UnitX());
+      transformA = AngleAxis3(angleDist(rng), Vector3::UnitY());
+      transformA = AngleAxis3(angleDist(rng), Vector3::UnitZ());
+
+      transformA.translation() << globDist(rng), globDist(rng), globDist(rng);
+
+      auto planeSurfaceA = Surface::makeShared<PlaneSurface>(transformA);
+
+      auto planeSurfaceB = Surface::makeShared<PlaneSurface>(transformA);
+
+      auto boundToBound =
+          [&gctx, &bField](
+              const auto& parIn, const auto& covIn, const auto& srfA,
+              const auto& srfB) -> std::pair<BoundVector, BoundMatrix> {
+        Acts::FreeVector freePars =
+            Acts::detail::transformBoundToFreeParameters(srfA, gctx, parIn);
+        Acts::BoundVector parOut =
+            Acts::detail::transformFreeToBoundParameters(freePars, srfB, gctx)
+                .value();
+
+        Acts::BoundToFreeMatrix boundToFreeJacobian =
+            srfA.boundToFreeJacobian(gctx, parIn);
+        Acts::FreeMatrix freeTransportJacobian = FreeMatrix::Identity();
+
+        FreeVector freeToPathDerivatives = FreeVector::Zero();
+        freeToPathDerivatives.head<3>() = freePars.segment<3>(eFreeDir0);
+
+        freeToPathDerivatives.segment<3>(eFreeDir0) =
+            bField.cross(freePars.segment<3>(eFreeDir0));
```

Comment:
@beomki-yeo I'm setting the direction derivative to $B\times T$ here.

Reply:
Should be OK for this test